### PR TITLE
Replaced "ceph-volume" calls with "cephadm ceph-volume" (#450)

### LINKED ISSUE #450: Change all instances of "ceph-volume" to "cephadm ceph-volume"
The ceph-volume executable is owned by the ceph-osd package. Since this package is not (currently) installed on the host system in SES7, and also not included in SES7 Media1, it is not possible to run ceph-volume in a SES7 cluster.

The workaround is:

1. Using "podman ps", determine the SHA1 of one of the running ceph containers
2. "podman exec -it [CONTAINER_SHA1] /bin/bash" to fire up a bash shell in the container
3. run ceph-volume commands, e.g. "ceph-volume inventory" works from inside the container

you'll need to run this via cephadm:

sudo ./cephadm ceph-volume inventory
INFO:cephadm:/usr/bin/podman:stderr Trying to pull docker.io/ceph/daemon-base:latest-master-devel...
INFO:cephadm:/usr/bin/podman:stderr Getting image source signatures
INFO:cephadm:/usr/bin/podman:stderr Copying blob sha256:ab5ef0e5819490abe86106fd9f4381123e37a03e80e650be39f7938d30ecb530
INFO:cephadm:/usr/bin/podman:stderr Copying blob sha256:de9856aacf1f5416e5ed0797172ac725228d7162d6718cc54a1081fe2eede5f2
INFO:cephadm:/usr/bin/podman:stderr Copying config sha256:60d27201f1f163f5f4f4714eb8062b068337acddb8b3ecec011ce990740f9681
INFO:cephadm:/usr/bin/podman:stderr Writing manifest to image destination
INFO:cephadm:/usr/bin/podman:stderr Storing signatures
INFO:cephadm:/usr/bin/podman:stdout 
INFO:cephadm:/usr/bin/podman:stdout Device Path               Size         rotates available Model name
INFO:cephadm:/usr/bin/podman:stdout /dev/nvme0n1              476.94 GB    False   False     KXG50ZNV512G NVMe TOSHIBA 512GB
INFO:cephadm:/usr/bin/podman:stdout /dev/sda                  1.82 TB      True    False     ST2000LM007-1R81

Device Path               Size         rotates available Model name
/dev/nvme0n1              476.94 GB    False   False     KXG50ZNV512G NVMe TOSHIBA 512GB
/dev/sda                  1.82 TB      True    False     ST2000LM007-1R81

This needs proper documentation

### PR DESCRIPTION
**Pull request by tbazant**
Changes the invocation of `ceph-volume` to run from inside containers via `cephadm`
Closes #450

### CODE REVIEW

**Comment by smithfarm** [comment]
This is fine as far as it goes, but I would go one step farther and, at the first instance in the Admin Guide where you mention "cephadm ceph-volume", insert a helpful, explanatory box. The text in the box should explain what happened that we are suddenly prefixing all ceph-volume calls with `cephadm `. Perhaps something as simple as this:

    NOTE: In SES7, the ceph-volume executable resides in a container, so it is not runnable directly.
    Use `cephadm` to run it by prefixing your `ceph-volume` command line with `cephadm `.

**Comment by tbazant** [comment]
> This is fine as far as it goes, but I would go one step farther and, at the first instance in the Admin Guide where you mention "cephadm ceph-volume", insert a helpful, explanatory box. The text in the box should explain what happened that we are suddenly prefixing all ceph-volume calls with `cephadm `. Perhaps something as simple as this:
> 
> ```
> NOTE: In SES7, the ceph-volume executable resides in a container, so it is not runnable directly.
> Use "cephadm" to run it by prefixing your "ceph-volume" command line with "cephadm ".
> ```

@smithfarm Does this approach apply to other commands as well?

**Comment by smithfarm** [comment]
> @smithfarm Does this approach apply to other commands as well?

Not that I'm aware of.

**Comment by tbazant** [comment]
> > @smithfarm Does this approach apply to other commands as well?
> 
> Not that I'm aware of, but there are many, many changes in the CLI user interface between SES6 and SES7, as a consequence of the move to containers. This is just one of them.

In the admin guide, there is no `ceph-volume` command :-) they're spread across all 5 guides. So I have to include that statement in some common section

**Comment by smithfarm** [comment]
Each guide is a separate, standalone book, so my suggestion would be to include the same statement in each book where the "cephadm ceph-volume" appears. It should always appear at the first instance of "cephadm ceph-volume" within the given book.

We cannot expect users to switch from book to book just to find explanatory notes.

**Review by lproven — APPROVED** [review]

**Comment by tbazant** [comment]
> Each guide is a separate, standalone book, so my suggestion would be to include the same statement in each book where the "cephadm ceph-volume" appears. It should always appear at the first instance of "cephadm ceph-volume" within the given book.
> 
> We cannot expect readers to switch from book to book just to find explanatory notes.

@smithfarm i've added the snippet to the starting common part of each book as agreed with Alex. i cannot gues what is the first occurrence of a string as it may change in time + if a customer googles it, it may well point them to the middle of a chapter. This PR seems huge as i had to restructure and align all our books accordingly

File: xml/MAIN.susestorage.xml [hunk 1]
```diff
@@ -41,8 +43,8 @@
...
 <!--SUSE Storage Security Hardening Guide-->
  <xi:include href="book_storage_security.xml" parse="xml"/>
- <!--Deploying and Administrating Ceph on SUSE CaaS Platform-->
-  <xi:include href="book_storage_rook.xml" parse="xml"/>
- <!--SUSE Storage Ceph for Windows Guide-->
-  <xi:include href="book-ceph-windows.xml" parse="xml"/>
+<!--Deploying and Administrating Ceph on SUSE CaaS Platform-->
+ <xi:include href="book_storage_rook.xml" parse="xml"/>
+<!--SUSE Storage Ceph for Windows Guide-->
+ <xi:include href="book-ceph-windows.xml" parse="xml"/>
```
Comment (smithfarm) [inline xml/MAIN.susestorage.xml:49]: why is the whitespace change needed?

Comment (smithfarm) [reply xml/MAIN.susestorage.xml:49]: oh, nvm, I see now

File: xml/ceph_prompts.xml [hunk 2]
```diff
@@ -117,23 +107,33 @@
...
-  <title>General Linux Commands</title>
+  <sect3>
+   <title>Running &ceph; Commands inside Containers</title>
+   <para>
+    Several executables&mdash;such as <command>ceph-volume</command>&mdash;
+    reside inside a running container. You cannot run them directly from the
+    underlying operating system, but via the <command>cephadm</command> module,
+    for example:
+   </para>
+<screen>&prompt.cephuser;cephadm ceph-volume simple scan</screen>
```
Comment (smithfarm) [inline xml/ceph_prompts.xml:118]: I'm not sure `cephadm ceph-volume`, which is a special case, is a good example. Other executables that are available only in the container image would be run using `cephadm shell`, which uses a different syntax.

Also, I don't think the wording "reside inside a running container" is correct. When the user issues a command like `cephadm ceph-volume` or `cephadm shell`, cephadm executes podman, which spawns a container from the ceph base container image. The commands actually reside inside this container image, not inside "a running container". The container doesn't actually run long enough to say something "resides" in it. But maybe I'm splitting hairs here.

Maybe @sebastian-philipp could suggest a better wording?

File: xml/common_intro_available_doc_i.xml [hunk 3]
```diff
@@ -31,46 +30,57 @@
    <term><xref linkend="book-storage-deployment"/></term>
    <listitem>
     <para>
-     Guides you through the installation steps of the &ceph; cluster and all
-     services related to &ceph;. The guide also illustrates a basic &ceph;
-     cluster structure and provides you with related terminology.
+     This guide focuses on deploying a basic &ceph; cluster and describes how
+     to deploy additional services as well. It also cover steps to upgrade to
```
Comment (smithfarm) [inline xml/common_intro_available_doc_i.xml:34]: It also covers steps (missing "s" in "covers")

File: xml/common_intro_cephwin.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+   <dm:priority/>
+   <dm:translation>yes</dm:translation>
+   <dm:languages/>
+   <dm:release>SES 7</dm:release>
+  </dm:docmanager>
+ </info>
+ <xi:include href="common_intro_product.xml" parse="xml"/>
+ <para>
+  This guide describes the integration, installation and configuration of
+  Microsoft Windows environments and &productname; susing the Windows Driver.
```
Comment (smithfarm) [inline xml/common_intro_cephwin.xml:25]: "using" instead of "susing"

**Comment by smithfarm** [comment]
I left some comments, but I'm afraid there's too much going on in this PR for my feeble knowledge of docbook/XML to comprehend.

File: xml/MAIN.susestorage.xml (hunk 1 above)
Comment (tbazant) [reply xml/MAIN.susestorage.xml:49]: just n XML formatting

File: xml/common_intro_available_doc_i.xml (hunk 3 above)
Comment (tbazant) [inline xml/common_intro_available_doc_i.xml:34]:
```suggestion
     to deploy additional services as well. It also covers steps to upgrade to
```

File: xml/common_intro_cephwin.xml (hunk 4 above)
Comment (tbazant) [inline xml/common_intro_cephwin.xml:25]:
```suggestion
  Microsoft Windows environments and &productname; using the Windows Driver.
```

**Review by tbazant — COMMENTED** [review]
2 of 3 suggestions taken into account

**Comment by tbazant** [comment]
> I left some comments, but I'm afraid there's too much going on in this PR for my feeble knowledge of docbook/XML to comprehend.

The structural changes are not relevant for you, the content is.

File: xml/ceph_prompts.xml [hunk 5]
```diff
@@ -117,23 +107,33 @@
...
   </tip>
- </sect1>
- <sect1>
-  <title>General Linux Commands</title>
+  <sect3>
+   <title>Running &ceph; Commands inside Containers</title>
+   <para>
+    Several executables&mdash;such as <command>ceph-volume</command>&mdash;
+    reside inside a running container. You cannot run them directly from the
+    underlying operating system, but via the <command>cephadm</command> module,
```
Comment (lproven) [inline xml/ceph_prompts.xml:115]:
```suggestion
    underlying operating system, but instead via the <command>cephadm</command> module,
```

File: xml/common_intro_available_doc_i.xml [hunk 6]
```diff
@@ -31,46 +30,57 @@
...
+     &productname; &productnumber; from the previous product version.
```
Comment (lproven) [inline xml/common_intro_available_doc_i.xml]:
```suggestion
     This guide focuses on deploying a basic &ceph; cluster, and how to 
     deploy additional services. It also cover the steps for upgrading to
     &productname; &productnumber; from the previous product version.
```

File: xml/common_intro_available_doc_i.xml [hunk 7]
```diff
@@ -31,46 +30,57 @@
...
+     that the user has a detailed plan on how to tune their system.
+    </para>
+   </listitem>
+  </varlistentry>
+  <varlistentry>
+   <term><xref linkend="book-ceph-windows"/></term>
+   <listitem>
+    <para>
+      This guide describes the integration, installation and configuration of
+      Microsoft Windows environments and &productname; susing the Windows Driver.
```
Comment (lproven) [inline xml/common_intro_available_doc_i.xml:83]:
```suggestion
      Microsoft Windows environments and &productname; using the Windows Driver.
```

File: xml/common_intro_cephwin.xml (hunk 4 above)
Comment (lproven) [inline xml/common_intro_cephwin.xml]:
```suggestion
  Microsoft Windows environments and &productname; using the Windows Driver.
```

File: xml/common_intro_deploy.xml [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+   <dm:translation>yes</dm:translation>
+   <dm:languages/>
+   <dm:release>SES 7</dm:release>
+  </dm:docmanager>
+ </info>
+ <xi:include href="common_intro_product.xml" parse="xml"/>
+ <para>
+  This guide focuses on deploying a basic &ceph; cluster and describes how to
+  deploy additional services as well. It also cover steps to upgrade to
+  &productname; &productnumber; from the previous product version.
```
Comment (lproven) [inline xml/common_intro_deploy.xml:26]:
```suggestion
  This guide focuses on deploying a basic &ceph; cluster, and how to
  deploy additional services. It also covers the steps for upgrading to
  &productname; &productnumber; from the previous product version.
```

File: xml/common_intro_rook.xml [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+ </info>
+ <xi:include href="common_intro_product.xml" parse="xml"/>
+ <para>
+  &rook; allows you to run &ceph; on &kube; to provide container workloads all
+  their storage needs. Part of the &productname; family is the &rook;
+  deployment which runs on &casp;.
+ </para>
+ <para>
+  The &rook; deployment is currently only available to nominated and approved
+  customers, it's in limited availability. Contact your SUSE sales team on how
```
Comment (lproven) [inline xml/common_intro_rook.xml:30]:
```suggestion
  customers,: it is in <emphasis>limited availability</emphasis>. For information on how
```

File: xml/common_intro_rook.xml [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+  to get nominated.
```
Comment (lproven) [inline xml/common_intro_rook.xml:31]:
```suggestion
  to get nominated, contact your SUSE sales team.
```

Comment (lproven) [reply xml/common_intro_rook.xml:31]: Also I suspect we should use an entity for "SUSE".

File: xml/rook-contributors.xml [hunk 11]
```diff
@@ -20,11 +20,13 @@
...
   </dm:docmanager>
  </info>
+
  <para>
-  The &ceph; and &rook; projects and their documentation is a result of the work
-  of hundreds of contributors and organizations. See
-  <link xlink:href="https://ceph.com/contributors/"/> for more details on &ceph;
-  and <link xlink:href="https://rook.io/docs/rook/v1.4/development-flow.html"/>
-  for more details on &rook;.
+  The &ceph; and &rook; projects and their documentation is a result of the
```
Comment (lproven) [inline xml/rook-contributors.xml:25]:
```suggestion
  The &ceph; and &rook; projects and their documentation are the result of the
```

**Review by lproven — CHANGES_REQUESTED** [review]

File: xml/ceph_prompts.xml [hunk 12]
```diff
@@ -117,23 +107,32 @@
...
   </tip>
- </sect1>
- <sect1>
-  <title>General Linux Commands</title>
+  <sect3>
+   <title>Running <command>ceph-volume</command></title>
+   <para>
+    Since &productname; 7, &ceph; services are running containerized. If you
+    need to run <command>ceph-volume</command> on an OSD node, you need to
+    prepend it with <command>cephadm</command>, for example:
```
Comment (tbazant) [inline xml/ceph_prompts.xml:115]:
```suggestion
    prepend it with the <command>cephadm</command> command, for example:
```

File: xml/ceph_prompts.xml [hunk 13]
```diff
@@ -117,23 +107,32 @@
 &prompt.cephuser.mon;
 </screen>
   </tip>
- </sect1>
- <sect1>
-  <title>General Linux Commands</title>
+  <sect3>
+   <title>Running <command>ceph-volume</command></title>
+   <para>
+    Since &productname; 7, &ceph; services are running containerized. If you
```
Comment (tbazant) [inline xml/ceph_prompts.xml:113]:
```suggestion
    Starting with &productname; 7, &ceph; services are running containerized. If you
```

**Pull request merged**